pkg/piecestore/psserver/psdb/psdb_test.go

```diff
@@ -274,6 +276,52 @@ func TestBandwidthUsage(t *testing.T) {
 	})
 }
 
+func TestMigration(t *testing.T) {
+	_, err := exec.LookPath("sqlite3")
+	if err != nil {
+		t.Skip("unable to find sqlite3 executable")
+	}
+
+	// find latest version
+	migration := psdb.Migration()
+	migration.Now = func() time.Time { return time.Unix(0, 0) }
+
+	latestVersion := 0
+	// TODO support missing intermediate versions
+	for _, step := range migration.Steps {
+		if latestVersion < step.Version {
+			latestVersion = step.Version
+		}
+	}
+
+	for version := 0; version <= latestVersion; version++ {
+		t.Run("migration-v"+strconv.Itoa(version), func(t *testing.T) {
+			ctx := testcontext.New(t)
+			defer ctx.Cleanup()
+
+			dbpath := ctx.File("psdb.db")
+			db, err := psdb.Open(dbpath)
+			require.NoError(t, err)
+			defer ctx.Check(db.Close)
+
+			testedMigration := migration.TargetVersion(version)
+			err = testedMigration.Run(zap.NewNop(), db)
+			require.NoError(t, err)
+
+			out, err := exec.Command("sqlite3", dbpath, ".dump").Output()
```

Comment:
Maybe there's a way to do this part with queries.

Reply:
For a moment was looking for that but most of the results give `.dump`. I will search once again before merging.